text/0000-hierarchic-anonymous-life-time.md

```diff
@@ -0,0 +1,136 @@
+- Feature Name: Hierarchic anonymous life-time
+- Start Date: 2020-06-24
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+New use of anonymous life-time `'_` that implicitly added to current structure.
+
+# Motivation
+
+Motivation is to simplify iterative development and improving refactoring of the code
+
+Sometimes during refactoring such code:
+```rust
+struct CompositeObject {
+    obj: SomeType,
+}
+
+struct BigObject {
+    composite_obj: CompositeObject,
+    count: i32,
+}
+
+struct Application {
+   big_obj: BigObject,
+}
```

Comment:
```suggestion
struct Country {
    name: String,
}

struct World {
    country: Country,
    count: i32,
}

struct Application {
   country: Country,
}
```

Too many confusing terms are used here (`obj`, `object`), I believe it should could be something more understandable with a good example.